File: DiffOddEvenFact/Diff.go
/*
	Author : Pakshal Jain
	Date : 17/01/2023
	Program : To Calculate Difference Between Even and Odd Factorial of Entered Number
*/

//Required Libraies
package main
import "fmt"

//////////////////////////////////////////////////////////////////////////////////////////////

//Function To Calculate Difference
func Difference(No int) int {
	var i int = 0;
	var EvenFact int = 1;
	var OddFact int = 1;
	var Diff int = 0;

	if No < 0 {
		No = -No;
	}

	//Logic To Calcualate Difference
	for i = No;i > 0;i-- {
		if i % 2 != 0 {
			OddFact = OddFact * i;
		} else {
			EvenFact = EvenFact * i;
		}
	}

	//Calculating Difference
	Diff = EvenFact - OddFact;

	return Diff
}

///////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
func main() {
	var No int = 0;
	var iret int = 0;

	fmt.Println("Jay Ganesh...");

	//Taking Input From User
	fmt.Println("Enter Number You Want : ");
	fmt.Scanf("%d",&No);

	//Call To Differnec Function
	iret = Difference(No);

	//Printing Obtained Function
	fmt.Println("Differnce Between Even and Odd Factorial of Entered Number is : ",iret);
}

File: DiffOddEvenFact/Diff.java
/*
    Author : Pakshal Jain
    Date : 17/01/2023
    Program : To Calculate Difference Between Even and Odd Factorial of Entered Number
*/

//Required Libraries
import java.lang.*;
import java.util.*;

///////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class Difference 
{
    //Characteristics of Class
    private int No = 0;

    //Constructor of Class Parameterised 
    Difference(int no)
    {
        if(no < 0)
        {
            no = -no;
        }

        this.No = no;
    }

    //Behaviour of Class
    int Calculate()
    {
        int EvenFact = 1;
        int OddFact = 1;
        int Diff = 0;
        int i = 0;

        //Logic To Calculate Difference
        for(i = this.No;i > 0;i--)
        {
            if(i % 2 != 0)
            {
                OddFact = OddFact * i;
            }
            else 
            {
                EvenFact = EvenFact * i;
            }
        }

        Diff = EvenFact - OddFact;

        return Diff;
    }
}//End of Class

////////////////////////////////////////////////////////////////////////////

//Main Class
class Diff
{
    //Main Function
    public static void main(String args[])
    {
        int No = 0;
        int iret = 0;
        Scanner sobj = new Scanner(System.in);

        System.out.println("Jay Ganesh...");

        //Taking Input From User
        System.out.println("Enter Any Number You Want : ");
        No = sobj.nextInt(); 

        //Creating object of Difference Class
        Difference dobj = new Difference(No);   

        //Call To Behaviour of Class
        iret = dobj.Calculate();

        //Printing Obtained Result
        System.out.println("Difference Between Even And Odd Factorial is : "+iret);
    }
}